### ISSUE TITLE
[Request] Standalone executables available for linux/arm64 and MasOS/arm64

### ISSUE THREAD

**Issue by lunrenyi** [hpjansson/chafa#223]
Hi, Chafa is an awesome tool! I found a [standalone executable for the x86_64 architecture](https://hpjansson.org/chafa/download/), but I couldn't find one for arm64.
Would you be able to provide **standalone executables for linux/arm64 and MasOS/arm64**?
![image](https://github.com/user-attachments/assets/0c2aa7ac-1c28-47a8-87ab-0f6e515a4347)

**Thank you for sharing the chafa! :)**

**Comment by hpjansson** [comment]
I can do linux/arm64 without too much trouble, but MacOS is harder as I have no Mac to build/test it on.

Would linux/arm64 be sufficient?

**Comment by GuardKenzie** [comment]
GitHub does provide ARM and x86 macOS runners so it might be possible to build those using GitHub Actions

**Comment by hpjansson** [comment]
> GitHub does provide ARM and x86 macOS runners so it might be possible to build those using GitHub Actions

Good point. I think that should be tried first. Producing a statically linked binary requires some trickery (i.e. building the whole stack including glib and its deps, and maybe patching out some subsystems we don't need), but it'd be a relief to have it running in a proper build env.

**Comment by lalyos** [comment]
+1 as right now If I try to install via brew on mac (arm64) I get an error:
```
Error: chafa: Failed to download resource "libx11"
Download failed: https://www.x.org/archive/individual/lib/libX11-1.8.12.tar.gz
```

**Comment by hpjansson** [comment]
> +1 as right now If I try to install via brew on mac (arm64) I get an error:
> ```
> Error: chafa: Failed to download resource "libx11"
> Download failed: https://www.x.org/archive/individual/lib/libX11-1.8.12.tar.gz
> ```

We don't depend on X11, so you may be able to ignore that dependency. Could be a good idea to notify the package maintainer - if ImageMagick is still listed as a dependency, it should be removed.

**Comment by hpjansson** [comment]
Well, we have a linux-armv7l binary available to test now, but this is 32-bit and intended for the RPi and maybe others: https://github.com/hpjansson/chafa/issues/308.

There's also a CoreGraphics loader coming in #301 that may be of interest to Mac users.

ARM64 was a little more trouble than I thought. We really need to look at a GH runner.
